source/darwin/secure_transport_tls_channel_handler.c

```diff
@@ -922,72 +925,42 @@ static struct aws_tls_ctx *s_tls_ctx_new(struct aws_allocator *alloc, struct aws
                 &secure_transport_ctx->certs)) {
             AWS_LOGF_ERROR(
                 AWS_LS_IO_TLS, "static: failed to import certificate and private key with error %d.", aws_last_error());
-            aws_secure_zero(cert_chain.buffer, cert_chain.len);
-            aws_byte_buf_clean_up(&cert_chain);
-            aws_secure_zero(private_key.buffer, private_key.len);
-            aws_byte_buf_clean_up(&private_key);
             goto cleanup_wrapped_allocator;
         }
-
-        aws_secure_zero(cert_chain.buffer, cert_chain.len);
-        aws_byte_buf_clean_up(&cert_chain);
-        aws_secure_zero(private_key.buffer, private_key.len);
-        aws_byte_buf_clean_up(&private_key);
-    } else if (options->pkcs12_path) {
-
-        struct aws_byte_buf pkcs12_blob;
-        if (aws_byte_buf_init_from_file(&pkcs12_blob, alloc, options->pkcs12_path)) {
-            AWS_LOGF_ERROR(AWS_LS_IO_TLS, "static: failed to load %s", options->pkcs12_path);
-            goto cleanup_wrapped_allocator;
-        }
-
-        struct aws_byte_buf password = {.buffer = NULL, .len = 0, .allocator = NULL, .capacity = 0};
-
-        if (options->pkcs12_password) {
-            password = aws_byte_buf_from_c_str(options->pkcs12_password);
-        }
-
-        struct aws_byte_cursor pkcs12_blob_cur = aws_byte_cursor_from_buf(&pkcs12_blob);
-        struct aws_byte_cursor password_cur = aws_byte_cursor_from_buf(&password);
+    } else if (options->pkcs12.len) {
+        struct aws_byte_cursor pkcs12_blob_cur = aws_byte_cursor_from_buf(&options->pkcs12);
```

Comment:
Debug log message?

Reply:
the logging is now in the file_utils. I don't know what i'd log here.

Reply:
For all the other branches, you AWS_LOGF_DEBUG that you're initializing

Reply:
oh, damn github diffs, I thought this was a different c file. will do